ui/core/proto_utils/stats.ts

```diff
@@ -470,6 +470,8 @@ export class Stats {
 			statDelta /= this.getPseudoStat(PseudoStat.PseudoStatRangedSpeedMultiplier);
 		} else if (unitStat.equalsPseudoStat(PseudoStat.PseudoStatSpellHastePercent)) {
 			statDelta /= this.getPseudoStat(PseudoStat.PseudoStatCastSpeedMultiplier);
+		} else if (unitStat.equalsPseudoStat(PseudoStat.PseudoStatAttackSpeedMultiplier)) {
+			statDelta /= this.getPseudoStat(PseudoStat.PseudoStatAttackSpeedMultiplier);
```

Comment:
I guess if you're adding this logic already, then it's fine to export this as an additional `PseudoStat`. Up to you whether you think it's valuable to separate it out in the character stats export, versus just combining it into the other three multipliers at the time of export. If you do decide to add it though, then rename it to `PseudoStatHasteMultiplier` for clarity, just like the other naming changes I suggested in the back-end.